### ISSUE TITLE
Redrawing in an event handler

### ISSUE THREAD

**Issue by mbouaziz** [caleb531/jcanvas#88]
I have a layer with .click = function F(layer) { canvas.setLayer(layer, {visible: false}); }
However the layer vanishes only once the mouse have moved.
But calling drawLayers() in F seems pretty bad for the stack.

Solutions:
- set a layer property 'redraw' to true in F so that drawLayer could redraw it
- any call to drawLayers in F is deferred to the return of F
